bears/general/LineLengthBear.py

```diff
@@ -1,26 +1,43 @@
 import re
+import logging
 
 from coalib.bearlib import deprecate_settings
 from coalib.bearlib.spacing.SpacingHelper import SpacingHelper
 from coalib.bears.LocalBear import LocalBear
 from coalib.results.Result import Result
 from coalib.settings.Setting import typed_list
+from coalib.bearlib.languages.Language import Language, UnknownLanguageError
+from coalib.bearlib.aspects import map_setting_to_aspect
+from coalib.bearlib.aspects.Formatting import (
+    Formatting,
+    LineLength,
+)
 
 
-class LineLengthBear(LocalBear):
+class LineLengthBear(
+        LocalBear,
+        aspects={
+            'detect': [
+                LineLength,
+            ]},
+        ):
     LANGUAGES = {'All'}
     AUTHORS = {'The coala developers'}
     AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
     LICENSE = 'AGPL-3.0'
     CAN_DETECT = {'Formatting'}
 
+    @map_setting_to_aspect(
+        max_line_length=LineLength.max_line_length,
+    )
     @deprecate_settings(indent_size='tab_width')
     def run(self,
             filename,
             file,
             max_line_length: int = 79,
             indent_size: int = SpacingHelper.DEFAULT_TAB_WIDTH,
             ignore_length_regex: typed_list(str) = (),
+            language: str = None,
```

Comment:
This should be `language: language = languages.Unknown`.  Also this arg should be the first setting, immediately after `file`. 
 
Was there some problem doing that?

I notice that `AnnotationBear` and `IndentationBear` uses `language: str`

`CPDBear` uses `language: language`

Reply:
The problem I am facing with `language: language` is in case of an invalid language. It is designed for bears where one valid language is compulsory. If the language is invalid, it jumps out of the bear with ValueError (without running the bear), we won't be able to set the language to `Unknown` if there is an invalid language. Even if I do `language: language = None`, it doesn't help.

Reply:
If the language is invalid, it should give a ValueError.  They should omit it, or specify `language = Unknown` ;-).